rules/embeds/image_compare.test.js

```diff
@@ -0,0 +1,69 @@
+</div>
+`);
+});
+
+it('handles image compare embeds with no urls (no embed)', () => {
+    expect(md.render('[image_compare  ]')).toBe(`<p>[image_compare  ]</p>
+`);
+});
+
+it('handles image compare embeds that are unclosed (no embed)', () => {
+    expect(md.render('[image_compare https://rb.gy/jykhuo https://rb.gy/zt5afg')).toBe(`<p>[image_compare https://rb.gy/jykhuo https://rb.gy/zt5afg</p>
+`);
+});
+
+it('handles image compare embeds without width', () => {
+    expect(md.render('[image_compare https://rb.gy/jykhuo https://rb.gy/zt5afg 400]')).toBe(`<div class="image-compare" style="--value:50%; height: 400px; width: 500px;">
+    <img class="image-left" src="https://rb.gy/jykhuo" alt="Image left"/>
+    <img class="image-right" src="https://rb.gy/zt5afg" alt="Image right"/>
+    <input type="range" class="control" min="0" max="100" value="50" oninput="this.parentNode.style.setProperty('--value', \`\${this.value}%\`)" />
+    <svg class="control-arrow" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M504.3 273.6c4.9-4.5 7.7-10.9 7.7-17.6s-2.8-13-7.7-17.6l-112-104c-7-6.5-17.2-8.2-25.9-4.4s-14.4 12.5-14.4 22l0 56-192 0 0-56c0-9.5-5.7-18.2-14.4-22s-18.9-2.1-25.9 4.4l-112 104C2.8 243 0 249.3 0 256s2.8 13 7.7 17.6l112 104c7 6.5 17.2 8.2 25.9 4.4s14.4-12.5 14.4-22l0-56 192 0 0 56c0 9.5 5.7 18.2 14.4 22s18.9 2.1 25.9-4.4l112-104z"/></svg>
+</div>
+`);
+});
+
+it('handles image compare embeds without width or height', () => {
+    expect(md.render('[image_compare https://rb.gy/jykhuo https://rb.gy/zt5afg]')).toBe(`<div class="image-compare" style="--value:50%; height: 500px; width: 500px;">
+    <img class="image-left" src="https://rb.gy/jykhuo" alt="Image left"/>
+    <img class="image-right" src="https://rb.gy/zt5afg" alt="Image right"/>
+    <input type="range" class="control" min="0" max="100" value="50" oninput="this.parentNode.style.setProperty('--value', \`\${this.value}%\`)" />
+    <svg class="control-arrow" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M504.3 273.6c4.9-4.5 7.7-10.9 7.7-17.6s-2.8-13-7.7-17.6l-112-104c-7-6.5-17.2-8.2-25.9-4.4s-14.4 12.5-14.4 22l0 56-192 0 0-56c0-9.5-5.7-18.2-14.4-22s-18.9-2.1-25.9 4.4l-112 104C2.8 243 0 249.3 0 256s2.8 13 7.7 17.6l112 104c7 6.5 17.2 8.2 25.9 4.4s14.4-12.5 14.4-22l0-56 192 0 0 56c0 9.5 5.7 18.2 14.4 22s18.9 2.1 25.9-4.4l112-104z"/></svg>
+</div>
+`);
+});
+
+it('handles image compare embeds attempting html injection', () => {
+    expect(md.render('[image_compare <script>alert();</script>]')).toBe(`<p>[image_compare &lt;script&gt;alert();&lt;/script&gt;]</p>
+`);
+});
+
+it('handles vimeo embeds attempting url manipulation', () => {
+    expect(md.render('[vimeo a/../../b 280 560]')).toBe(`<p>[vimeo a/../../b 280 560]</p>
+`);
+});
```

Comment:
This test is not testing the correct plugin

Reply:
Fixed.